From e7a8dfe92ef2dd136ed0ef435482c7717c08cd56 Mon Sep 17 00:00:00 2001
From: Martin Fick <mfick@codeaurora.org>
Date: Wed, 25 Aug 2010 14:51:12 -0600
Subject: [PATCH] Split several functions in MyWatchedProjectsScreen

Refactor two functions out of deleteChecked():

 remove(Set<AccountProjectWatch.Key> ids)

and

 Set<AccountProjectWatch.Key> getCheckedIds()

This should help clarify how deleteChecked() works and make
maintenance a bit easier since it is now possible to override
each of the methods separately. Also factor out the
populateWatches() function.

Change-Id: Ia1bff8e628be5ed8e911b23b0b2e7bc1133d12f6
---
 .../client/account/MyWatchedProjectsScreen.java    |   66 ++++++++++++--------
 1 files changed, 40 insertions(+), 26 deletions(-)

diff --git a/gerrit-gwtui/src/main/java/com/google/gerrit/client/account/MyWatchedProjectsScreen.java b/gerrit-gwtui/src/main/java/com/google/gerrit/client/account/MyWatchedProjectsScreen.java
index af02739..18b192e 100644
--- a/gerrit-gwtui/src/main/java/com/google/gerrit/client/account/MyWatchedProjectsScreen.java
+++ b/gerrit-gwtui/src/main/java/com/google/gerrit/client/account/MyWatchedProjectsScreen.java
@@ -45,6 +45,7 @@ import com.google.gwtjsonrpc.client.VoidResult;
 
 import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 
 public class MyWatchedProjectsScreen extends SettingsScreen {
   private WatchTable watches;
@@ -144,6 +145,12 @@ public class MyWatchedProjectsScreen extends SettingsScreen {
     add(delSel);
   }
 
+  @Override
+  protected void onLoad() {
+    super.onLoad();
+    populateWatches();
+  }
+
   void doAddNew() {
     final String projectName = nameTxt.getText();
     if ("".equals(projectName)) {
@@ -182,16 +189,14 @@ public class MyWatchedProjectsScreen extends SettingsScreen {
         });
   }
 
-  @Override
-  protected void onLoad() {
-    super.onLoad();
-    Util.ACCOUNT_SVC
-        .myProjectWatch(new ScreenLoadCallback<List<AccountProjectWatchInfo>>(
-            this) {
-          public void preDisplay(final List<AccountProjectWatchInfo> result) {
-            watches.display(result);
-          }
-        });
+  protected void populateWatches() {
+    Util.ACCOUNT_SVC.myProjectWatch(
+        new ScreenLoadCallback<List<AccountProjectWatchInfo>>(this) {
+      @Override
+      public void preDisplay(final List<AccountProjectWatchInfo> result) {
+        watches.display(result);
+      }
+    });
   }
 
   private class WatchTable extends FancyFlexTable<AccountProjectWatchInfo> {
@@ -220,31 +225,40 @@ public class MyWatchedProjectsScreen extends SettingsScreen {
     }
 
     void deleteChecked() {
-      final HashSet<AccountProjectWatch.Key> ids =
-          new HashSet<AccountProjectWatch.Key>();
-      for (int row = 1; row < table.getRowCount(); row++) {
-        final AccountProjectWatchInfo k = getRowItem(row);
-        if (k != null && ((CheckBox) table.getWidget(row, 1)).getValue()) {
-          ids.add(k.getWatch().getKey());
-        }
-      }
+      final Set<AccountProjectWatch.Key> ids = getCheckedIds();
       if (!ids.isEmpty()) {
         Util.ACCOUNT_SVC.deleteProjectWatches(ids,
             new GerritCallback<VoidResult>() {
               public void onSuccess(final VoidResult result) {
-                for (int row = 1; row < table.getRowCount();) {
-                  final AccountProjectWatchInfo k = getRowItem(row);
-                  if (k != null && ids.contains(k.getWatch().getKey())) {
-                    table.removeRow(row);
-                  } else {
-                    row++;
-                  }
-                }
+                remove(ids);
               }
             });
       }
     }
 
+    void remove(Set<AccountProjectWatch.Key> ids) {
+      for (int row = 1; row < table.getRowCount();) {
+        final AccountProjectWatchInfo k = getRowItem(row);
+        if (k != null && ids.contains(k.getWatch().getKey())) {
+          table.removeRow(row);
+        } else {
+          row++;
+        }
+      }
+    }
+
+    Set<AccountProjectWatch.Key> getCheckedIds() {
+      final Set<AccountProjectWatch.Key> ids =
+          new HashSet<AccountProjectWatch.Key>();
+      for (int row = 1; row < table.getRowCount(); row++) {
+        final AccountProjectWatchInfo k = getRowItem(row);
+        if (k != null && ((CheckBox) table.getWidget(row, 1)).getValue()) {
+          ids.add(k.getWatch().getKey());
+        }
+      }
+      return ids;
+    }
+
     void insertWatch(final AccountProjectWatchInfo k) {
       final String newName = k.getProject().getName();
       int row = 1;
-- 
1.7.2.2

